src/main/java/io/jenkins/plugins/gitlabserverconfig/servers/GitLabServer.java

```diff
@@ -242,6 +250,17 @@ public String getHooksRootUrl() {
         return Util.ensureEndsWith(Util.fixEmptyAndTrim(hooksRootUrl), "/");
     }
 
+    @DataBoundSetter
+    public void setSecretToken(String token) {
+        this.secretToken = token;
+    }
+
+    private void generateSecretToken() {
+        byte[] random = new byte[16];   // 16x8=128bit worth of randomness, since we use md5 digest as the API token
+        RANDOM.nextBytes(random);
+        this.secretToken = Util.toHexString(random);
+    }
+
```

Comment:
This us unused.

Reply:
Yes, will be used by some UI element to trigger this method, thats a TODO on me.